### PR TITLE
Feature/link camera text recognition

### PR DESCRIPTION
**Pull request by edoulaX**

### CODE REVIEW

File: app/src/androidTest/java/com/github/sdpsharelook/TextDetectionActivityTest.kt [hunk 1]
```diff
@@ -55,58 +48,23 @@ class TextDetectionActivityTest {
...
-        val textView2 = onView(
-            allOf(
-                withId(R.id.text_data),
-                withParent(withParent(withId(android.R.id.content))),
-                isDisplayed()
-            )
-        )
-        textView2.check(matches(isDisplayed()))
+        }
     }
```
Comment (AdrianUlrich) [inline app/src/androidTest/java/com/github/sdpsharelook/TextDetectionActivityTest.kt:68]: This is a huge red flag

File: app/src/main/java/com/github/sdpsharelook/textDetection/TextDetection.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                OnSuccessListener<Text?> { texts ->
+                    textResult = texts.text
+                })
+            .addOnFailureListener(
+                OnFailureListener { e -> // Task failed with an exception
+                    e.printStackTrace()
+                }).isComplete){
+            return textResult
+        }
+        return textResult
```
Comment (AdrianUlrich) [inline app/src/main/java/com/github/sdpsharelook/textDetection/TextDetection.kt:26]: very bad handling of asynchronous code: the task is likely to never be complete
make your function `suspend` and use `Task.await()`

File: app/src/main/java/com/github/sdpsharelook/textDetection/TextDetectionActivity.kt [hunk 3]
```diff
@@ -28,84 +28,22 @@ class TextDetectionActivity : AppCompatActivity() {
...
-//                for (element in line.elements) {
-//                    val elementText = element.text
-//                    val elementCornerPoints = element.cornerPoints
-//                    val elementFrame = element.boundingBox
-//                }
-//            }
-//        }
-    }
-
-
```
Comment (AdrianUlrich) [inline app/src/main/java/com/github/sdpsharelook/textDetection/TextDetectionActivity.kt:47]: nice simplification

File: app/src/main/java/com/github/sdpsharelook/TranslateActivity.kt [hunk 4]
```diff
@@ -63,6 +67,10 @@ class TranslateActivity : AppCompatActivity() {
             val intent = Intent(ctx, NavigationMenuActivity::class.java)
             ctx.startActivity(intent)
         }
+        if(translateWord){
+            sourceText.text = intent.getStringExtra(TRANSLATOR_WORD)
+            translateWord = false
+        }
```
Comment (AdrianUlrich) [inline app/src/main/java/com/github/sdpsharelook/TranslateActivity.kt:73]: wtf? there has to be a better way X(

**Review by AdrianUlrich — CHANGES_REQUESTED** [review]
I'm a bit shocked by how brutal your code is. But as long as it works...
Also please fix: think of the poor souls out there trying to convert the activities to fragments

**Review by AdrianUlrich — APPROVED** [review]
I approve because it will be a nice bughunt for later

File: app/src/main/java/com/github/sdpsharelook/TranslateActivity.kt (hunk 4 above)
Comment (edoulaX) [reply app/src/main/java/com/github/sdpsharelook/TranslateActivity.kt:73]: Yes, I was thinking to ask you how should we do this in a better way as we are going to do it for a few other activities.

Comment (albastardoto) [reply app/src/main/java/com/github/sdpsharelook/TranslateActivity.kt:73]: Probably should add a public function that can be called from other activities to set the translate text and translate it.
That way you can start the activity and call the function

**Review by albastardoto — APPROVED** [review]

**Pull request merged**